umh-core/pkg/communicator/actions/benthos-log-checker.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2025 UMH Systems GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package actions
+
+import (
+	"strings"
+
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/service/s6"
+)
+
+var ErrBenthosLogLines = []string{
+	"unable to infer",
+	"Config lint error",
+}
+
+// CheckBenthosLogLinesForConfigErrors examines a slice of S6 log entries for
+// Benthos configuration errors. It performs case-insensitive checks for known
+// error patterns combined with the word "error". Returns true if any errors
+// are detected, false otherwise.
+func CheckBenthosLogLinesForConfigErrors(logs []s6.LogEntry) bool {
```

Comment:
Can you explain here why this function is useful? 

--> When benthos starts up, it can land up in a CrashLoop because of a configuration error. THese manifest through ErrBenthosLogLines. So instead of forcing the user to wait for 20 more seconds, we use this function to immediately abort because if one of these issues appears it will require a configuration change